File: Nickvision.Application.GNOME/Controls/ShortcutsDialog.cs
using Nickvision.Application.GNOME.Helpers;
using Nickvision.Desktop.Globalization;

namespace Nickvision.Application.GNOME.Controls;

public class ShortcutsDialog
{
    private readonly Gtk.Builder _builder;
    private readonly Adw.ShortcutsDialog _shortcutsDialog;

    public ShortcutsDialog(ITranslationService translator)
    {
        _builder = Gtk.Builder.NewFromBlueprint("ShortcutsDialog", translator);
        _shortcutsDialog = new Adw.ShortcutsDialog(new Adw.Internal.ShortcutsDialogHandle(_builder.GetPointer("root"), false));
    }

    public void Present(Gtk.Widget? parent) => _shortcutsDialog.Present(parent);
}

File: org.nickvision.application.linux/src/controls/shortcuts_dialog.cpp
#include "controls/shortcuts_dialog.h"

using namespace application::linux::helpers;
using namespace desktop::app;

namespace application::linux::controls
{
	shortcuts_dialog::shortcuts_dialog(const std::shared_ptr<translation_service>& translation_service)
	    : m_builder{ "shortcuts_dialog", translation_service },
	      m_dialog{ m_builder.get<AdwShortcutsDialog>("root") }
	{
	}

	AdwShortcutsDialog* shortcuts_dialog::gobj() const
	{
		return m_dialog;
	}

	void shortcuts_dialog::present(GtkWidget* parent)
	{
		adw_dialog_present(ADW_DIALOG(m_dialog), parent);
	}
}